tests/snmp/test_snmp_interfaces.py

```diff
@@ -10,6 +12,85 @@
 
 logger = logging.getLogger(__name__)
 
+SAI_PORT_STAT_IF_IN_ERRORS = 'SAI_PORT_STAT_IF_IN_ERRORS'
+SAI_PORT_STAT_IF_OUT_ERRORS = 'SAI_PORT_STAT_IF_OUT_ERRORS'
+SAI_PORT_STAT_IF_IN_DISCARDS = 'SAI_PORT_STAT_IF_IN_DISCARDS'
+SAI_PORT_STAT_IF_OUT_DISCARDS = 'SAI_PORT_STAT_IF_OUT_DISCARDS'
+SAI_ROUTER_INTERFACE_STAT_IN_ERROR_PACKETS = 'SAI_ROUTER_INTERFACE_STAT_IN_ERROR_PACKETS'
+SAI_ROUTER_INTERFACE_STAT_OUT_ERROR_PACKETS = 'SAI_ROUTER_INTERFACE_STAT_OUT_ERROR_PACKETS'
+
+COUNTERS_PORT_NAME_MAP = 'COUNTERS_PORT_NAME_MAP'
+COUNTERS_RIF_NAME_MAP = 'COUNTERS_RIF_NAME_MAP'
+COUNTER_VALUE = 5000
+
+
+@pytest.fixture()
+def disable_conterpoll(duthost):
+    """
+    Disable conterpoll for RIF and PORT and re-enable it when TC finished
+    :param duthost: DUT host object
+    :return: dict with data collected from DUT per each port
+    """
+    ConterpollHelper.disable_counterpoll(duthost, counter_type_list=['port', 'rif'])
+    yield
+    ConterpollHelper.enable_counterpoll(duthost, counter_type_list=['port', 'rif'])
+
+
+def get_interfaces(duthost, tbinfo):
+    """
+    Method to get interfaces for testing
+    :param duthost: DUT host object
+    :return: RIF interface name in case available in topo. If not - return Port Channel name and interface in Port
+     Channel
+    """
+    rif_counters = parse_rif_counters(duthost.command("show interfaces counters rif")["stdout_lines"])
+    for interface in rif_counters:
+        if 'Eth' in interface:
+            return interface, interface
+        else:
+            mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+            return mg_facts["minigraph_portchannels"][interface]['members'][0], interface
+
+
+def get_oid_for_interface(duthost, table_name, interface_name):
+    """
+    Method to get interface oid from Counters DB
+    :param duthost: DUT host object
+    :param table_name: table name
+    :param interface_name: interface name
+    :return: oid for specific interface
+    """
+    return duthost.command(f"docker exec -i database redis-cli --raw -n 2 HMGET "
+                           f"{table_name} {interface_name}")["stdout"]
+
+
+def set_counters_value(duthost, interface_oid, counter_name, counter_value):
+    """
+    Method to set interface counter value in Counters DB
+    :param duthost: DUT host object
+    :param interface_oid: oid value
+    :param counter_name: counter name
+    :param counter_value: counter value
+    """
+    counters_oid = "COUNTERS:{}".format(interface_oid)
+    duthost.command(f"sudo redis-cli -n 2 hset {counters_oid} {counter_name} {counter_value}")
```

Comment:
this will not work as expected for multi-asic, use above suggestion

Reply:
Thank you @SuvarnaMeenakshi, will fix your comments. How about "I do see There is a cross-feature dependence. File: tests/snmp/test_snmp_interfaces.py, import module: tests.platform_tests.counterpoll.counterpoll_helper

how community deal with such problem , as I need already created function from counterpoll and do not want to duplicate or break existing logic. Can you share some example? Thank you in advance."

Reply:
Hi @SavchukRomanLv , I see that you have already imported the required module in this PR, I will share if I find any example of import other than from tests/common since that is commonly used across.
For the counterpoll specifically, I see this being used in conftest in test/snmp:
https://github.com/sonic-net/sonic-mgmt/blob/15d48b825e10df2534e32a6ff1505ab56c1a4049/tests/snmp/conftest.py#L99
Can you check if this is something that you can use or modify?

Reply:
Hi @SuvarnaMeenakshi  please check, fixed acc to your comments